app/models/search.rb

```diff
@@ -11,42 +11,45 @@ class Search
   validates :term, presence: true
   validates :dob,
             presence: true,
-            if: proc { |search| search.filter.eql?('defendant') }
+            if: proc { |search| search.filter.eql?('defendant_name') }
 
   def filters
     self.class.filters
   end
 
   def self.filters
     [
-      SearchFilter.new(
-        id: :case_reference,
-        name: I18n.t('search_filter.radio_case_reference_label'),
-        description: I18n.t('search_filter.radio_case_reference_label_hint')
-      ),
-      SearchFilter.new(
-        id: :defendant,
-        name: I18n.t('search_filter.radio_defendant_label'),
-        description: I18n.t('search_filter.radio_defendant_label_hint')
-      )
+      SearchFilter.new(id: :case_reference,
+                       name: I18n.t('search_filter.radio_case_reference_label'),
+                       description: I18n.t('search_filter.radio_case_reference_label_hint')),
+      SearchFilter.new(id: :defendant_reference,
+                       name: I18n.t('search_filter.radio_defendant_reference_label'),
+                       description: I18n.t('search_filter.radio_defendant_reference_label_hint')),
+      SearchFilter.new(id: :defendant_name,
+                       name: I18n.t('search_filter.radio_defendant_name_label'),
+                       description: I18n.t('search_filter.radio_defendant_name_label_hint'))
     ]
   end
 
   def execute
-    adaptor.call
+    query.call
   end
 
   private
 
-  def adaptor
-    send("#{filter}_adaptor")
+  def query
+    send("#{filter}_query")
```

Comment:
since `filter` is being set via user input, would it be advisable to restrict the values it can take?

Reply:
will look into, although the value is set by radio options in the view and there is no injection vector to be concerned with, i believe, since any other value will just simply raise a NoMethodError (which will filter up to a 500)

- but could validate against search filter collection ids.

Reply:
have added a basic validation against the filter attribute now, cheers